SSA/Projects/InstCombine/LLVM/Semantics.lean

```diff
@@ -8,48 +8,82 @@ import SSA.Projects.InstCombine.LLVM.SimpSet
 
 namespace LLVM
 
+
+abbrev IntW w := Option <| BitVec w
+
 /--
 The ‘and’ instruction returns the bitwise logical and of its two operands.
 -/
 @[simp_llvm]
-def and? {w : Nat} (x y : BitVec w) : Option <| BitVec w :=
+def and? {w : Nat} (x y : BitVec w) : IntW w :=
   some <| x &&& y
 
+@[simp_llvm_option]
+def and {w : Nat} (x y : IntW w) : IntW w := do
+  let x' ← x
+  let y' ← y
+  and? x' y'
```

Comment:
Do we ever use `and?` outside of the definition of `and`?
I would prefer just inlining `and?` (idem for `or?`, `xor?`, etc)

Reply:
There are uses of sdiv/udiv/srem/urem autside of the non-questionmark function. We also have two simp-sets, one for eliminating the Option interface and one for eliminating our definitions and falling to BitVec. I feel for the current development this might be useful.